### PR TITLE
Build and Test

### PR DESCRIPTION
**Pull request by andrewkatson**
I tried to get it building and testing. The two things I noticed that probably can be fixed but Im not sure how are 

1. Need to get `unit_test_main` linking properly (in other modules adding `linkstatic=True` was enough but not here)
2. Need to figure out why the different letter files are causing linking issues -- e.g. A.hpp and A.ipp

### CODE REVIEW

**Comment by zaucy** [comment]
I added bazel CI if you want to update your branch!

**Comment by andrewkatson** [comment]
Just did it!

File: BUILD.bazel [hunk 1]
```diff
@@ -8,8 +8,9 @@ cc_library(
         "include/**/*.hpp",
         "include/**/*.h",
         "include/**/*.ipp",
-    ]),
+    ]) ,
```
Comment (zaucy) [inline BUILD.bazel]: hey I've been noticing a few formatting issues in some of the build files you're sending occasionally. Are you using `buildifier`? If not then I'd highly recommend setting it up in your editor!

Comment (andrewkatson) [reply BUILD.bazel]: I am not using that. I will figure out how to set that up in VS Code!

**Review by zaucy — APPROVED** [review]
looking good! is this ready to merge?

**Comment by andrewkatson** [comment]
Yes I believe so.

**Pull request merged**